charts/redpanda/templates/statefulset.yaml

```diff
@@ -68,28 +68,58 @@ spec:
               mountPath: /var/lib/redpanda/data
         - name: {{ template "redpanda.name" . }}-configurator
           image: {{ .Values.image.repository}}:{{ template "redpanda.tag" . }}
+          # command: ["/bin/bash", "-c"]
           command: ["/bin/sh", "-c"]
           env:
             - name: SERVICE_NAME
               valueFrom:
                 fieldRef:
                   fieldPath: metadata.name
+            - name: EXTERNAL_NODES
+              value: {{ .Values.external.nodes }}
           args:
             - >
               CONFIG=/etc/redpanda/redpanda.yaml;
               cp /tmp/base-config/redpanda.yaml "$CONFIG";
+              NODE_ID=${SERVICE_NAME##*-};
               {{- if (include "redpanda-atleast-22-1-1" . | fromJson).bool }}
               cp /tmp/base-config/bootstrap.yaml /etc/redpanda/.bootstrap.yaml;
               {{- end }}
               {{- if (include "redpanda-atleast-22-3-0" . | fromJson).bool }}
               rpk --config "$CONFIG" config set redpanda.empty_seed_starts_cluster false;
               {{- else }}
-              NODE_ID=${SERVICE_NAME##*-};
               rpk --config "$CONFIG" config set redpanda.node_id $NODE_ID;
               if [ "$NODE_ID" = "0" ]; then
                 rpk --config "$CONFIG" config set redpanda.seed_servers '[]' --format yaml;
               fi;
               {{- end }}
+              rpk redpanda --config "$CONFIG" config set redpanda.advertised_kafka_api[0].name internal ;
+              rpk redpanda --config "$CONFIG" config set redpanda.advertised_kafka_api[0].address {{ $advertiseAddress }} ;
+              rpk redpanda --config "$CONFIG" config set redpanda.advertised_kafka_api[0].port {{ .Values.listeners.kafka.port }} ;
```

Comment:
Why are we doing this for internal?

Reply:
We don't actually configure any `advertised_kafka_api` in the configmap, it was originally in this same file as a startup flag. I considered moving all this `advertised_kafka_api, but like seed servers this configuration depends on the node ID (and the configmap is identical across all nodes).